modules/@apostrophecms/area/ui/apos/apps/AposAreas.js

```diff
@@ -23,9 +24,23 @@ export default function() {
   });
 
   apos.bus.$on('refreshed', function() {
+    /* killObsoleteApps(); */
     createAreaAppsAndRunPlayersIfDone();
   });
 
+  /* function killObsoleteApps() { */
```

Comment:
Good point about cleaning up the apps as well as the components. Given how messy this is, I think you would be better off using:

`this.$parent`

To access the app from the component's refreshed handler, and then you can destroy the app rather than the component, as it should recursively destroy the component for you. Worth a try!